File: code_learning/牛客/contest/2024-7-17/H.py
import sys

input = lambda: sys.stdin.readline().rstrip()


n = int(input())
a = list(map(int, input().split()))
ans = sum(a)
num1 = max(a) * (n - 1)
num2 = n
ans += num2 if num1 <= num2 else num1
# if num1 <= num2:
#     ans += num2
# else:
#     ans += num1

sys.stdout.write(str(ans))

File: code_learning/牛客/contest/2024-7-17/题解/H.cpp
#include "bits/stdc++.h"
using namespace std;
int main()
{
    int n;
    cin >> n;
    long long mx = 0, sum = 0;
    vector<int> a(n + 1);
    for (int i = 1; i <= n; i++)
    {
        cin >> a[i];
        sum += a[i];
    }
    for (int i = 1; i <= n; i++)
    {
        mx = max({mx, sum - a[i] + 1ll * a[i] * n, sum - a[i] + a[i] + n});
    }
    cout << mx << '\n';
}